internal/infrastructure/kubernetes/infra_resource.go

```diff
@@ -324,8 +338,30 @@ func (i *Infra) createOrUpdateHPA(ctx context.Context, r ResourceRender) (err er
 		}
 	)
 
+	defer func() {
+		if hpa != nil {
+			deleteErr := i.Client.DeleteAllExcept(ctx, &autoscalingv2.HorizontalPodAutoscalerList{}, client.ObjectKey{
+				Namespace: hpa.Namespace,
+				Name:      hpa.Name,
+			}, &client.ListOptions{
+				Namespace:     hpa.Namespace,
+				LabelSelector: r.LabelSelector(),
+			})
+			if deleteErr != nil {
+				i.logger.Error(deleteErr, "failed to delete all except HorizontalPodAutoscaler",
+					"name", r.Name(), "namespace", r.Namespace())
+			}
+		}
+
+		if err == nil {
```

Comment:
should probably be first above L342

Reply:
it's fine, because we use another `deleteErr`.